### PR TITLE
Added `rust_common.create_crate_info`

### PR DESCRIPTION
**Pull request by UebelAndre**
The `rust_common.create_crate_info` macro creates a `CrateInfo` provider through a macro which allows for the rules to maintain better backwards compatibility when making changes to `CrateInfo`.

### CODE REVIEW

File: rust/private/common.bzl [hunk 1]
```diff
@@ -25,7 +25,11 @@ In the Bazel lingo, `rust_common` gives the access to the Rust Sandwich API.
 
 load(":providers.bzl", "CrateInfo", "DepInfo")
 
+def _create_crate_info(**kwargs):
+    return CrateInfo(**kwargs)
+
 rust_common = struct(
+    create_crate_info = _create_crate_info,
```
Comment (hlopko) [inline rust/private/common.bzl:32]: I know it's not exported to https://bazelbuild.github.io/rules_rust/, but could you also add a doc saying that create_crate_info should be used for construction, and crate_info for accessing?

**Review by hlopko — APPROVED** [review]

File: rust/private/common.bzl (hunk 1 above)
Comment (UebelAndre) [reply rust/private/common.bzl:43]: Like, a docstring? (like the one I added?)

Comment (hlopko) [reply rust/private/common.bzl:43]: Oh right, yeah that works. Nit: rename "macro" to "function" in the docstring

Comment (UebelAndre) [reply rust/private/common.bzl:43]: Is the terminology not "macro"? But updated.

**Pull request merged**